nservicebus/upgrades/7to8/dependency-injection.md

```diff
@@ -0,0 +1,103 @@
+
+The following snippet configures NServiceBus to use the Microsoft's default container implementation. This requires the `Microsoft.Extensions.DependencyInjection` NuGet package.
+
+```csharp
+var serviceCollection = new ServiceCollection();
+var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
+
+var endpoint = await startableEndpoint.Start(serviceCollection.BuildServiceProvider());
+```
+
+#### Autofac
+
+The following snippet configures NServiceBus to use Autofac as its dependency injection container. This requires the `Autofac.Extensions.DependencyInjection` NuGet package.
+
+```csharp
+var serviceCollection = new ServiceCollection();
+var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
+
+var containerBuilder = new ContainerBuilder();
+containerBuilder.Populate(serviceCollection);
+var autofacContainer = containerBuilder.Build();
+var endpointInstance = await startableEndpoint.Start(new AutofacServiceProvider(autofacContainer));
+```
+
+#### StructureMap
+
+Note: StructureMap has been sunsetted. The maintainers recommend to use [Lamar](https://jasperfx.github.io/lamar/) instead.
+
+The following snippet configures NServiceBus to use StructureMap as its dependency injection container. This requires the `StructureMap.Microsoft.DependencyInjection` NuGet package.
+
+```csharp
+var serviceCollection = new ServiceCollection();
+var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
+
+var container = new Container();
+container.Populate(serviceCollection);
+var endpointInstance = await startableEndpoint.Start(container.GetInstance<IServiceProvider>());
+```
+
+
+#### Castle Windsor
+
+The following snippet configures NServiceBus to use Castle Windsor as its dependency injection container. This requires the `Castle.Windsor.MsDependencyInjection` NuGet package.
+
+```csharp
+var serviceCollection = new ServiceCollection();
+var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
+
+var container = new WindsorContainer();
+var serviceProvider = WindsorRegistrationHelper.CreateServiceProvider(container, serviceCollection);
+var endpointInstance = await startableEndpoint.Start(serviceProvider);
+```
+
+#### Unity
+
+The following snippet configures NServiceBus to use Unity as its dependency injection container. This requires the `Castle.Windsor.MsDependencyInjection` NuGet package.
+
+```csharp
+var serviceCollection = new ServiceCollection();
+var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
+
+var container = new WindsorContainer();
+var serviceProvider = WindsorRegistrationHelper.CreateServiceProvider(container, serviceCollection);
+var endpointInstance = await startableEndpoint.Start(serviceProvider);
+```
```

Comment:
@timbussmann @mauroservienti the references to `Castle.Windsor` look like a copy paste error?

Reply:
um, yeah it does 🤦

Reply:
raised https://github.com/Particular/docs.particular.net/pull/5304